x/cosmwasmpool/README.md

```diff
@@ -274,36 +274,15 @@ pub enum SudoMessage {
 
 ## Incentives and Shares
 
-In order to allow CosmWasm pool to work with incentives module (or being composable in general), the contract needs to be able to create shares token and have ability to mint/burn them.
+In order to allow CosmWasm pool to work with incentives module (or being composable in general),
+the contract needs to be able to create share tokens.
 
-Each pool have share denom with this pattern: `cosmwasmpool/address/{contract_address}`.
+We handle this by utilizing the `x/tokenfactory` module.
+Each pool have share denom with this pattern: `tokenfactory/{contract_address}/{pool_id}`.
```

Comment:
- I think existing pattern for tokenfactory is `factory/{creator_address}/{subdenom}`, it can't start with `tokenfactory/...`
- Do we need to specify it at all? With this we can leave how the contract handle their own shares mechanism flexible unless there is a need to.

Reply:
yeah, if the contract address already uniquely identifies a pool, then we should be able to let the contract use any name they want. 

What we could do is add a string prefix (say, `.../cosmwasmpool/`) so that the denoms are easily recognizable as shares.

Reply:
> I think existing pattern for tokenfactory is factory/{creator_address}/{subdenom}, it can't start with tokenfactory/...

Copied from tokenfactory README but I will confirm and correct both

> Do we need to specify it at all? With this we can leave how the contract handle their own shares mechanism flexible unless you have reasons that I'm unaware of.

>yeah, if the contract address already uniquely identifies a pool, then we should be able to let the contract use any name they want.
> What we could do is add a string prefix (say, .../cosmwasmpool/) so that the denoms are easily recognizable as shares.

This is something I wasn't sure about and agreed with the proposal to use any name. Will update this

Reply:
Addressed [8e97a12](https://github.com/osmosis-labs/osmosis/pull/5350/commits/8e97a12a3d069b07e630979638433383875d799f)